test/e2e/secondary_network_test.go

```diff
@@ -0,0 +1,557 @@
+	if err != nil || respCode != 0 {
+		return fmt.Errorf("Error when deploying %s remote cluster - rc: %v, err: %v", pathOfWhereabouts, respCode, err)
+	}
+	return nil
+}
+
+func (data *TestData) deleteWhereaboutsCNI() error {
+	pathOfWhereabouts := "." + secondary_network_config_path + whereabouts
+	respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl delete -f %s -n kube-system", pathOfWhereabouts))
+	if err != nil || respCode != 0 {
+		return fmt.Errorf("Error when deleting %s remote cluster - rc: %v, err: %v", pathOfWhereabouts, respCode, err)
+	}
+	return nil
+}
+
+func (data *TestData) deployAntreaCNI() error {
+	pathOfAntreaCNI := "." + secondary_network_config_path + antrea
+	respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s -n kube-system", pathOfAntreaCNI))
+	if err != nil || respCode != 0 {
+		return fmt.Errorf("Error when deploying %s remote cluster - rc: %v, err: %v", pathOfAntreaCNI, respCode, err)
+	}
+	return nil
+}
+
+func (data *TestData) deleteAntreaCNI() error {
+	pathOfAntreaCNI := "." + secondary_network_config_path + antrea
+	respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl delete -f %s -n kube-system", pathOfAntreaCNI))
+	if err != nil || respCode != 0 {
+		return fmt.Errorf("Error when deleting %s remote cluster - rc: %v, err: %v", pathOfAntreaCNI, respCode, err)
+	}
+	return nil
+}
+
+// This function sets the IP link down of the provided Network interface
+func (data *TestData) toggleIPLinkAtNode(networkInterfaceName string, state int) error {
+	if state == 0 {
+		respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("ip link set %s down", networkInterfaceName))
+		if err != nil || respCode != 0 {
+			return fmt.Errorf("Toggle network interface '%s' state to UP at remote node failed!.", networkInterfaceName)
+		}
+	} else {
+		respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("ip link set %s up", networkInterfaceName))
+		if err != nil || respCode != 0 {
+			return fmt.Errorf("Toggle network interface '%s' state to DOWN at remote node failed!.", networkInterfaceName)
+		}
+	}
+	return nil
+}
+
+// This function forms the annotation string, used in the generation of each pod yaml file.
+func formAnnotationStringOfPod(pod int) string {
+	var annotation_str = ""
+	for _, xPodVN := range listOfVirtualNetworksPerPod {
+		var Input = "{\"name\": \"" + nameOfVirtualNetworkPerPod[pod][xPodVN] + "\" ,\"interface\": \"" + nameOfInterfacePerPod[pod][xPodVN] + "\" , \"type\": \"sriov\"}"
+		if annotation_str == "" {
+			annotation_str = "[" + Input
+		} else {
+			annotation_str = annotation_str + "," + Input
+		}
+	}
+	annotation_str = annotation_str + "]"
+	return annotation_str
+}
+
+func generatePodSpecWithSecondaryNetworkAnnotation() error {
```

Comment:
As we talked, I would create Pods with Go client, and maybe leverage & extend the existing Pod creation framework (e.g. CreatePodOnNodeInNamespace()). It is simpler and consistent with other tests, than implementing another way of using kubectl and YAML.

Reply:
Indeed I agree. Need some help here please. I couldn't find this suitable for the pod spec we need with Annotation. Let me review this again and check. will ping in slack after this.